### ISSUE TITLE
Reflection and auto-box warnings - wrt GraalVM

### ISSUE THREAD

**Issue by ingydotnet** [kkinnear/zprint#305]
From https://clojurians.slack.com/archives/CAJN79WNT/p1696690011308229

Ingy döt Net: Not sure where to ask about this but I started using zprint for the yamlscript compiler to nicely format clojure code.

When I run lein test it gives me a ton of warnings with  `*warn-on-reflection*`

```
Auto-boxing loop arg: column
zprint.cljc:5687 recur arg for primitive local: index is not matching primitive, had: Object, needed: long
Auto-boxing loop arg: index
zprint.cljc:7753 recur arg for primitive local: newline_count is not matching primitive, had: Object, needed: long 
Auto-boxing loop arg: newline-count
Reflection warning, zprint/zprint.cljc:8181:27 - reference to field toString can't be resolved. 
```

Asking here as I've been told that reflections are bad for graalvm native-image...

borkdude: Boxing doesn’t hurt GraalVM native image. The warnings you’re seeing here are because of warn-on-boxed. Only important if you want better perf, not important for getting it to work

Ingy döt Net: Well there is that

```
Reflection warning, zprint/zprint.cljc:8181:27 - reference to field toString can't be resolved.
```

line above...

Also kind of weird that warn-on-boxed is enabled. I only set `*warn-on-reflection*` (in project.clj)

borkdude: Perhaps warn-on-boxed is set inside of the project itself. The toString thing can be fixed in zprint

here: https://github.com/kkinnear/zprint/blob/ce5a1caf676b00f2b754e35151042211fbcc57f8/src/zprint/zprint.cljc#L8181

```
    (fzprint* options ind (. zloc toString))
<https://github.com/kkinnear/zprint|kkinnear/zprint>kkinnear/zprint | Added by GitHub
```

but if you add a reflection config for Object you can likely just work around it

better to fix it inside of zprint though

just replacing the interop by `(str ..)` will fix the issue

**Comment by kkinnear** [comment]
I suspect that I can change the form of that `toString` thing in the next release.  I don't actually know why it was done that way originally, but I haven't spent any time getting into it.  I've certainly had to work around plenty of graalVM based issues elsewhere in zprint, this would just be one more.

It is interesting that I don't see this error, as I use graalVM to build the executables that I distribute on GitHub.  But possibly it is the way I build it with graalVM, in that I tell it to go ahead and build it and complain only if it actually encounters a problem during execution.  I am sure that this particular piece of code isn't executed by my tests that put the graalVM images through their paces.  I'll see if I can provoke graalVM by some additional testing of the existing images.

Anyway, I'll work on making this better for you.  Thanks for reporting it!

**Comment by kkinnear** [comment]
I'm sorry to have taken so long to make progress on this, a number of personal issues have gotten in the way.

I have fixed two reflection issues in zprint, and now, interestingly, with the latest graalVM it builds without `--report-unsupported-elements-at-runtime`, which it did not before.  I have several times in the past fixed all of the reflection problems, but I don't routinely search for and remove them -- something that I will add to my release checklist for the future.

I havne't really done much development since the last release, so I'm not excited about a new release just now, for a variety of reasons.  That said, I would like to enable you to use zprint if this is blocking you from doing so.  Since the only substantive changes are these reflection issues (and a few boxing issues I worked to fix), I could fairly easily produce a new library, say `1.2.9-alpha1` at clojars that included these changes.  Of course, they would also be included in the final `1.2.9` as well as any future releases.  

Would that be of a help to you?
